src/storage/src/mut_records.rs

```diff
@@ -98,9 +122,40 @@ impl MutFileRecords {
         if self.f_sink.can_be_appended(buffer.len() as u64) {
             debug!("writing {} bytes at: {}", buffer.len(), self.path.display());
             self.f_sink.write_all(&buffer).await?;
-            // for now, we flush for every send
-            self.f_sink.flush().await?;
-            debug!("flushed {}", self.path.display());
+            match self.flush_policy {
+                NoFlush => {}
+
+                EveryWrite => {
+                    self.f_sink.flush().await?;
+                    debug!("EveryWrite flushed {}", self.path.display());
+                }
+
+                CountWrites {
+                    n_writes: n,
+                    count: icounts,
+                } => {
+                    let cnt = icounts + 1;
+                    if cnt >= n {
+                        self.flush_policy = CountWrites {
+                            n_writes: n,
+                            count: 0,
+                        };
+                        self.f_sink.flush().await?;
+                        debug!("CountWrites flushed {}", self.path.display());
+                    } else {
+                        self.flush_policy = CountWrites {
+                            n_writes: n,
+                            count: cnt,
```

Comment:
Should add increment method to `CountWrite` instead of allocation

Reply:
In the update, I did this inline to the count directly, as an increment method on enum would require a redundant match on the enum variant. I can still do it as a method if you think its cleaner.

Reply:
Yes, I think it would more clean  to separate out implementation

Reply:
ok, updated